dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/predictor/DefaultPredictionService.java

```diff
@@ -192,6 +199,44 @@ private List<Predictor> fetchPredictors( List<String> predictors, List<String> p
 
     @Override
     public void predict( Predictor predictor, Date startDate, Date endDate, PredictionSummary predictionSummary )
+    {
+        for ( Predictor p : preprocessor.preprocess( predictor ) )
+        {
+            // Trigger a new transaction when calling self (must be inside loop)
+            PredictionService self = applicationContext.getBean( PredictionService.class );
+
+            self.predictSimple( p, startDate, endDate, predictionSummary );
```

Comment:
I know why but it screams bad design :D
Lets put a short comment here also that we need to go through the proxy for transactions.

Reply:
I've expanded the comment to talk about proxy as well as transaction. Yeah, it does seem a bit convoluted.